hw_21/tests/test_bank.py

```diff
@@ -0,0 +1,87 @@
+"""Homework_21"""
+
+import pytest
+from hw_21.source.bank import Deposit, Bank
+from hw_21.my_logger import logger as log
+
+
+@pytest.fixture
+def deposit():
+    deposit = Deposit(1000, 12, 0.10)
+    yield deposit
+
+
+@pytest.fixture
+def bank():
+    bank = Bank()
+    yield bank
+
+
+def test_deposit_initialization(deposit):
+    assert deposit.amount == 1000
+    assert deposit.term == 12
+    assert deposit.percent == 0.10
+
+
+def test_deposit_total_amount(deposit):
+    expected_amount = 1000 * ((1 + 0.10 / 12) ** 12)
+    assert deposit.total_amount() == pytest.approx(expected_amount, rel=1e-4)
+
+
+def test_bank_deposit(bank):
+    N = 1000
+    R = 12
+    total_amount = bank.deposit(N, R)
+    expected_amount = 1000 * ((1 + 0.10 / 12) ** 12)
+    assert total_amount == pytest.approx(expected_amount, rel=1e-4)
```

Comment:
Can this test be parameterized?

```suggestion
@pytest.mark.parametrize(("amount", "months", "expected"), [(1000, 12, <value>), ....])
def test_bank_deposit(bank, amount, months, expected):
    total_amount = bank.deposit(N, R)
    assert total_amount == pytest.approx(expected_amount, rel=1e-4)
```

Reply:
sure, done. Thank you for suggestion :)